bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/tracer/BSpan.java

```diff
@@ -158,4 +171,24 @@ public void addTag(String tagKey, String tagValue) {
         }
         return carrierMap;
     }
+
+    public BMap<BString, Object> getBSpanContext() {
+
+        if (bSpanContext == null) {
+            SpanContext spanContext = span.getSpanContext();
+            if (spanContext.isSampled()) {
+                MapType immutableStringMap = TypeCreator.createMapType(PredefinedTypes.TYPE_STRING, true);
```

Comment:
Shall we make this a public static field as well ?

Reply:
changed